### PR TITLE
docbook-xsl: Make dbtoepub executable

### PR DESCRIPTION
**Pull request by petere**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This used to work, but somehow at the moment the `.../docbook-xsl/epub/bin/dbtoepub` file ends up without executable permissions, and the wrapper script fails when you try to call it.

### CODE REVIEW

**Comment by petere** [comment]
The ubuntu test is failing because ruby is not installed and it's used for the new test I added. Is it worth installing ruby for this test? Does this indicate that the package should have a dependency on ruby?

File: Formula/docbook-xsl.rb [hunk 1]
```diff
@@ -60,6 +60,7 @@ def install
       doc.install "doc" => "reference"
     end
 
+    chmod 0755, "#{prefix}/docbook-xsl/epub/bin/dbtoepub"
```
Comment (SMillerDev) [inline Formula/docbook-xsl.rb:63]: Why is this needed if we wrap it in the line below?

File: Formula/docbook-xsl.rb [hunk 2]
```diff
@@ -103,5 +104,6 @@ def post_install
     system "xmlcatalog", "#{etc}/xml/catalog", "http://docbook.sourceforge.net/release/xsl/#{version}/"
     system "xmlcatalog", "#{etc}/xml/catalog", "http://docbook.sourceforge.net/release/xsl-ns/current/"
     system "xmlcatalog", "#{etc}/xml/catalog", "http://docbook.sourceforge.net/release/xsl-ns/#{version}/"
+    system bin/"dbtoepub", "--help"
```
Comment (SMillerDev) [inline Formula/docbook-xsl.rb:107]: Is there no better test than help

File: Formula/docbook-xsl.rb (hunk 1 above)
Comment (petere) [reply Formula/docbook-xsl.rb:63]: The wrapper script will fail if the thing it calls it not executable:

    /usr/local/bin/dbtoepub: line 2: /usr/local/Cellar/docbook-xsl/1.79.2_1/docbook-xsl/epub/bin/dbtoepub: Permission denied
    /usr/local/bin/dbtoepub: line 2: exec: /usr/local/Cellar/docbook-xsl/1.79.2_1/docbook-xsl/epub/bin/dbtoepub: cannot execute: Undefined error: 0

File: Formula/docbook-xsl.rb (hunk 2 above)
Comment (petere) [reply Formula/docbook-xsl.rb:107]: Probably, but at the moment the challenge is to even get "--help" working. ;-)

I'll think of something.

File: Formula/docbook-xsl.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/docbook-xsl.rb:63]: Actually, if this should be executable it should be fixed upstream.

**Comment by iMichka** [comment]
> chmod 0755, "#{prefix}/docbook-xsl/epub/bin/dbtoepub"

> Actually, if this should be executable it should be fixed upstream.

Agreed. Using chmod here is the same as carrying our own patch around: we avoid to do this in general.
The fix needs to be done upstream. I am going to close this, unless there this is a Homebrew-only issue, then we can re-open this for discussion.

**Comment by petere** [comment]
But then why install a `dbtoepub` wrapper if it doesn't work?
